lib/ansible/galaxy/role.py

```diff
@@ -68,6 +68,9 @@ def __init__(self, galaxy, api, name, src=None, version=None, scm=None, path=Non
         self.scm = scm
         self.paths = [os.path.join(x, self.name) for x in galaxy.roles_paths]
 
+        if self.src.startswith('~'):
```

Comment:
Why should it be conditional?

Reply:
src can be an url and AFAIK, there is no case where ~ should be expanded when not at the start of the string. Basically I prefered to be safe than risking to mess some weird urls (not really expected in urls either).

Reply:
`os.path.expanduser()` never expands tilde unless at the beginning of the string:
```python
In [1]: import os.path                                                        

In [2]: os.path.expanduser('~')
Out[2]: '/home/wk'

In [3]: os.path.expanduser('/~')
Out[3]: '/~'

In [4]: os.path.expanduser('http://~')
Out[4]: 'http://~'
```
meaning there's no point in having this check